src/app/shared/components/carousel/carousel.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit, AfterContentInit } from '@angular/core';
+import { ViewChild, ElementRef } from '@angular/core';
+
+@Component({
+  selector: 'app-carousel',
+  templateUrl: './carousel.component.html',
+  styleUrls: ['./carousel.component.scss']
+})
+export class CarouselComponent implements OnInit, AfterContentInit {
+
+  @ViewChild('allSlidesContainer') allSlides: ElementRef;
```

Comment:
don't use an alias here. Rename to just slidesContainer


`@ViewChild()` slidesContainer: ElementRef;`

Reply:
by @ViewChild() argument between '' is required. 
<img width="1440" alt="2019-01-10 10 13 02" src="https://user-images.githubusercontent.com/25414782/50954897-5ee73d80-14c0-11e9-8619-d92a32229ce1.png">

Reply:
ok, just rename to slideContainer